### PR TITLE
KAFKA-5901: Added Connect metrics specific to source tasks

### PR DESCRIPTION
**Pull request by rhauch**
Added Connect metrics specific to source tasks, and builds upon #3864 and #3911 that have already been merged into `trunk`.

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/metrics/stats/Frequencies.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+     *                                  {@link Frequency#centerValue() center value} within the specified range
+     */
+    public Frequencies(int buckets, double min, double max, Frequency... frequencies) {
+        super(0.0);
+        if (max < min) {
+            throw new IllegalArgumentException("The maximum value " + max
+                                                       + " must be greater than the minimum value " + min);
+        }
+        if (buckets < 1) {
+            throw new IllegalArgumentException("Must be at least 2 buckets");
```
Comment (tedyu) [inline clients/src/main/java/org/apache/kafka/common/metrics/stats/Frequencies.java]: The message doesn't seem to match the condition above.

Comment (rhauch) [reply clients/src/main/java/org/apache/kafka/common/metrics/stats/Frequencies.java]: Good catch. Will fix shortly.

**Comment by wushujames** [comment]
Can you update the docs as well? docs/ops.html

**Comment by rhauch** [comment]
@wushujames, I'll follow up with the docs change in a separate PR.

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java [hunk 2]
```diff
@@ -421,4 +435,99 @@ public String toString() {
...
+    }
+
+    static class SourceTaskMetricsGroup {
+        private final MetricGroup metricGroup;
+        private final Sensor sourceRecordPoll;
+        private final Sensor sourceRecordWrite;
+        private final Sensor pollTime;
+
+        public SourceTaskMetricsGroup(ConnectorTaskId id, ConnectMetrics connectMetrics) {
+            metricGroup = connectMetrics.group("source-tasks-metrics",
```
Comment (ewencp) [inline connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java:458]: from the KIP, I think this is supposed to be `source-task-metrics`

**Review by ewencp — COMMENTED** [review]
i think this is ready to go except for the one nit. i can also clean it up on commit.

there was also the same issue re: commented stuff, though I'm not too worried about that

File: connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java (hunk 2 above)
Comment (rhauch) [reply connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java:458]: Ack

**Comment by ewencp** [comment]
Ok, based on ack, this LGTM and I'll just clean up that one extra `s` during merge.

**Pull request closed without merging**